File: app/src/main/java/com/example/energo_monitoring/checks/adapters/FlowTransducerPhotoAdapter.java
package com.example.energo_monitoring.checks.adapters;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.activity.result.ActivityResultLauncher;
import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.example.energo_monitoring.checks.presenters.CheckLengthOfStraightLinesPresenter;
import com.example.energo_monitoring.R;

public class FlowTransducerPhotoAdapter extends RecyclerView.Adapter<FlowTransducerPhotoAdapter.PhotoViewHolder> {

    CheckLengthOfStraightLinesPresenter presenter;
    ActivityResultLauncher<Intent> takePhotoLauncher;
    Context context;

    public FlowTransducerPhotoAdapter(Context context, CheckLengthOfStraightLinesPresenter presenter) {
        this.context = context;
        this.presenter = presenter;
    }

    public static class PhotoViewHolder extends RecyclerView.ViewHolder{
        public PhotoViewHolder(@NonNull View itemView) {
            super(itemView);
        }

        public void loadImageFromBitmap(Bitmap bitmap){
            ImageView photoImageView = itemView.findViewById(R.id.photoImageView);
            photoImageView.setImageBitmap(bitmap);
        }
    }

    @NonNull
    @Override
    public PhotoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.list_item_flow_transducers_photo_card, parent, false);

        PhotoViewHolder vh = new PhotoViewHolder(view);

        CardView cardView = view.findViewById(R.id.photoCard);
        cardView.setOnClickListener(v -> presenter.onPhotoCardClicked(vh));

        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull PhotoViewHolder holder, int position) {
        holder.loadImageFromBitmap(presenter.photos.get(position));
    }

    @Override
    public int getItemCount() {
        return presenter.photos.size();
    }

}

File: app/src/main/java/com/example/energo_monitoring/checks/ui/adapters/FlowTransducerPhotoAdapter.kt
package com.example.energo_monitoring.checks.ui.adapters

import androidx.recyclerview.widget.RecyclerView
import android.graphics.Bitmap
import com.example.energo_monitoring.R
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import androidx.cardview.widget.CardView
import com.example.energo_monitoring.checks.ui.vm.CheckLengthVM
import javax.inject.Inject

class FlowTransducerPhotoAdapter constructor(var viewModel: CheckLengthVM) :
    RecyclerView.Adapter<FlowTransducerPhotoAdapter.PhotoViewHolder>() {

    class PhotoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun loadImageFromBitmap(bitmap: Bitmap?) {
            val photoImageView = itemView.findViewById<ImageView>(R.id.photoImageView)
            photoImageView.setImageBitmap(bitmap)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotoViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.list_item_flow_transducers_photo_card, parent, false)

        val vh = PhotoViewHolder(view)
        val cardView = view.findViewById<CardView>(R.id.photoCard)
        cardView.setOnClickListener { viewModel.onPhotoCardClicked(vh.layoutPosition) }
        return vh
    }

    override fun onBindViewHolder(holder: PhotoViewHolder, position: Int) {
        holder.loadImageFromBitmap(viewModel.photos[position])
        if (position + 1 != itemCount) {
            holder.itemView.findViewById<ImageView>(R.id.photoImageView).visibility = View.INVISIBLE
        }
    }

    override fun getItemCount(): Int = viewModel.photos.size
}